### ISSUE TITLE
markdown support does not work within \if{latex}{} condition?

### ISSUE THREAD

**Issue by allenzhuaz** [r-lib/roxygen2#1048]
Hi, 
I really like the roxygen build markdown support with option **Roxygen: list(markdown = TRUE)** or **@md**, but when I wrote some documentation within the **if{latex}{}** condition, the markdown language did not work within the latex condition, _even if I set the markdown support option._ For example, I have to write multiple items as:
#' \if{latex}{
#'  Example:
#'  \itemize{
#'  \item 1st item
#'  \item 2nd item
#'  \item 3rd item }}

rather than
#' \if{latex}{
#'  Example:
#'  * 1st item
#'  * 2nd item
#'  * 3rd item }

Similarly, for inserting a table, seemingly I am only allowed to write:
#' \if{latex}{
#'  \tabular{ll}{
#'  a \tab b \cr
#'  c \tab d 
#' }}
but not allowed to use markdown, like:
#' \if{latex}{
#'  |  a  |  b  |
#'  |----|----|
#'  |  c  |  d  |
#' }

**Comment by gaborcsardi** [comment]
Maybe we can fix this. Can you provide a reproducible examples please? See https://github.com/r-lib/roxygen2/blob/master/.github/SUPPORT.md#making-a-reprex

**Comment by allenzhuaz** [comment]
Please see the reproducible example:

library(roxygen2)
roc_proc_text(rd_roclet(), "
#' @section Test
#' \if{latex}{
#' List example:
#' * 1st item
#' * 2nd item
#' * 3rd item
#' Table example:
#' | a | b |
#' |---|---|
#' | c | d |
#' }
#' @export
fun <- function(){return(NULL)}
")

**Comment by hadley** [comment]
Actual reprex:

``` r
library(roxygen2)
roc_proc_text(rd_roclet(), "
#' Title
#' @section Test:
#' \\if{latex}{
#' List example:
#' * 1st item
#' * 2nd item
#' * 3rd item
#' Table example:
#' | a | b |
#' |---|---|
#' | c | d |
#' }
#' @export
fun <- function(){return(NULL)}
")[[1]]
#> % Generated by roxygen2: do not edit by hand
#> % Please edit documentation in ./<text>
#> \name{fun}
#> \alias{fun}
#> \title{Title}
#> \usage{
#> fun()
#> }
#> \description{
#> Title
#> }
#> \section{Test}{
#> 
#> \if{latex}{
#> List example:
#> * 1st item
#> * 2nd item
#> * 3rd item
#> Table example:
#> | a | b |
#> |---|---|
#> | c | d |
#> }
#> }
```

<sup>Created on 2020-03-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by hadley** [comment]
I think getting this to work is going to be hard, because we'd have to parse the Rd in order to figure out which bits should be transformed.

**Comment by gaborcsardi** [comment]
According to my comments at https://github.com/r-lib/roxygen2/blob/357d856f6ccdb96b7a95b42bf747cf10e2a26fb3/R/markdown-escaping.R#L13-L16 this should work. 

But it does not. It is probably not _that_ hard, because we already escape these macros, so they are already parsed.

**Comment by gaborcsardi** [comment]
Yeah, so I think for this we need to add another mode to the `roxygen_parse_tag()` C function, to only parse the first argument of `\if` or `\ifelse` and leave the rest in place.

**Comment by allenzhuaz** [comment]
> Yeah, so I think for this we need to add another mode to the `roxygen_parse_tag()` C function, to only parse the first argument of `\if` or `\ifelse` and leave the rest in place.
> 
> https://github.com/r-lib/roxygen2/blob/e86cebf914331f580cc96c3f26a6cd203f82b1be/src/isComplete.cpp#L14

Thanks for looking into this feature, still on demand for the feature in our development, wondering whether it will be implemented in the next released version?

**Comment by gaborcsardi** [comment]
We'll look at this issue before the next release, yes.

**Comment by hadley** [comment]
Giving up on this because I now believe it would require a combined markdown + Rd parser, which I'm doubtful is worth the effort.
